### PR TITLE
Add first prototype of triggers as a service (TaaS)

### PR DESCRIPTION
**Pull request by cocreature**
This is an extremely basic version of the trigger as a service thingy.

Right now, it supports spawning triggers and stopping them but nothing
else.

There is a very simple test to check that it’s not completely broken.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+  }
+  implicit val triggerParamsFormat = jsonFormat2(TriggerParams)
+
+  def apply(
+      host: String,
+      port: Int,
+      ledgerConfig: LedgerConfig,
+      dar: Dar[(PackageId, Package)],
+  ): Behavior[Message] = Behaviors.setup { ctx =>
+    // implicit val system: ActorSystem = ctx.system
```
Comment (aherrmann-da) [inline triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala]:
```suggestion
```

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    // http doesn't know about akka typed so provide untyped system
+    implicit val untypedSystem: akka.actor.ActorSystem = ctx.system.toClassic
+    implicit val materializer: Materializer = Materializer(untypedSystem)
+    // implicit val ec: ExecutionContext = ctx.system.executionContext
```
Comment (aherrmann-da) [inline triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala]:
```suggestion
```

File: triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+}
+import com.digitalasset.platform.common.LedgerIdMode
+import com.digitalasset.platform.sandbox.SandboxServer
+import com.digitalasset.platform.sandbox.config.SandboxConfig
+import com.digitalasset.platform.services.time.TimeProviderType
+// import scalaz._
+// import scalaz.std.option._
+// import scalaz.std.scalaFuture._
+// import scalaz.syntax.traverse._
+
```
Comment (aherrmann-da) [inline triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala]: Can those be removed?

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+
+    var triggers: Map[UUID, ActorRef[TriggerActor.Message]] = Map.empty
+
+    val route = concat(
+      post {
+        // Start a new trigger given its identifier and the party it should be running as.
+        // Returns a UUID for the newly started trigger.
+        path("start") {
+          entity(as[TriggerParams]) { params =>
+            println(params)
```
Comment (aherrmann-da) [inline triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala]: Is this a debug left-over? Maybe use logger instead?

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceConfig.scala [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+package com.digitalasset.daml.lf.engine.trigger
+
+import java.nio.file.{Path, Paths}
+import java.time.Duration
+
+import com.digitalasset.platform.services.time.TimeProviderType
+
+case class ServiceConfig(
+    // For now, we only support one dar.
```
[the remark this answers is not in the document]
Comment (cocreature) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceConfig.scala]: Yep.

File: triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala (hunk 3 above)
Comment (cocreature) [reply triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala]: good catch, removed.

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala (hunk 4 above)
Comment (cocreature) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala]: I jut removed it, it wasn’t really useful for anything.

**Pull request merged**